controller.go

```diff
@@ -469,3 +457,71 @@ func upgradePVFrom12(pv *PV) (deleted bool, err error) {
 	}
 	return false, nil
 }
+
+func provisionClaim(pvc *api.PersistentVolumeClaim) error {
+	// Provisioning can take some time. Start it in a separate goroutine.
+	// Keep a map of running provisioners and start new provisioning only if
+	// it's not already running. There will be some locks around, these are
+	// left out to implementation
+
+	if !runningProvisioners.isRunning(pvc) {
```

Comment:
We cant tell if a provisioner is running, especially in the case of external / out of tree / independent provisioners.  We could use a provisionerStarted timestamp instead.